openedx/core/djangoapps/user_api/management/commands/sync_hubspot_contacts.py

```diff
@@ -103,13 +107,18 @@ def _get_batched_users(self, users, batch_size=100):
         """
         Splits user's list into batches
         Args:
-            users: list of users to be batched
+            users: users (generator) to be batched
             batch_size: size of batch
         """
-        total = len(users)
-        for start in range(0, total, batch_size):
-            end = min(start + batch_size, total)
-            yield (start, end, total, users[start:end])
+        end = 0
+        users_iter = iter(users)
+        while True:
+            users_batch = [user for _, user in itertools.izip(xrange(batch_size), users_iter)]
```

Comment:
NIT: This could be simplified using `islice` to `users_batch = list(itertools.islice(users_iter, batch_size))`